assets/src/controllers/campaign_form_controller.js

```diff
@@ -1,74 +1,564 @@
-import { Controller } from "stimulus";
-import _ from "lodash";
-import axios from "../utils/axios";
+import { Controller } from 'stimulus';
+import _ from 'lodash';
+import axios from '../utils/axios';
 
 export default class extends Controller {
   static get targets() {
-    return [
-      "ecpm",
-      "totalBudget",
-      "estimatedImpressions",
-      "includedCountries",
-      "creatives",
-      "userId"
-    ];
-  }
-
-  connect() {
-    this.calculateImpressions();
+    return ['ecpm', 'totalBudget', 'includedCountries', 'creatives', 'userId'];
   }
 
   creativesForUser(context) {
     axios({
       url: `/users/${context.srcElement.value}/creatives/index.json`,
-      method: "GET",
+      method: 'GET',
       headers: {
-        "content-type": "application/x-www-form-urlencoded"
-      }
-    })
-    .then(this.handleCreativesResponse.bind(this))
+        'content-type': 'application/x-www-form-urlencoded',
+      },
+    }).then(this.handleCreativesResponse.bind(this));
   }
 
   handleCreativesResponse(response) {
     var len = response.data.creatives.length;
     var options = _.map(response.data.creatives, function(creative) {
-      return '<option value="'+ creative.id +'">'+ creative.name +'</option>';
-    })
+      return (
+        '<option value="' + creative.id + '">' + creative.name + '</option>'
+      );
+    });
     this.creativesTarget.innerHTML = options;
   }
 
-  calculateImpressions() {
```

Comment:
@coderberry @hopsoft We need this field to calculate total impressions for a campaign so that the redis service has a number to compare for total budget. Super important, can't be removed now.